mi/reporting/resources.py

```diff
@@ -0,0 +1,73 @@
+from datetime import date, datetime
+from pathlib import Path
+from typing import Generator
+
+from helpers.log import log
+from mi.reporting.paths import PATH_TO_QUERIES
+
+RESOURCE_PREFIX = "nhsd-nrlf"
+LAMBDA_NAME = RESOURCE_PREFIX + "--{workspace}--mi--sql_query"
+SECRET_NAME = RESOURCE_PREFIX + "--{workspace}--{workspace}--{operation}_password"
+DB_CLUSTER_NAME = RESOURCE_PREFIX + "-{env}-aurora-cluster"
+
+
+@log("Got credentials")
+def get_credentials(session, workspace: str, operation: str = "read") -> dict:
+    secret_name = SECRET_NAME.format(workspace=workspace, operation=operation)
+    client = session.client("secretsmanager")
+    response = client.get_secret_value(SecretId=secret_name)
+    return {"user": f"{workspace}-{operation}", "password": response["SecretString"]}
+
+
+@log("Got SQL statement {__result__}")
+def each_sql_statement() -> Generator[tuple[str, str], None, None]:
+    for path in Path(PATH_TO_QUERIES).iterdir():
+        if path.is_dir():
+            continue
+        report_name = path.stem
+        with open(path) as f:
+            sql_statement = f.read()
+            yield report_name, sql_statement
+
+
+@log("Got endpoint {__result__}")
+def get_endpoint(session, env: str, operation: str = "read") -> str:
```

Comment:
Most of these comments are naming based sorry - but id suggest something like `get_rds_endpoint` or something to be clear over some other type of endpoint